src/core/handshaker/handshaker.cc

```diff
@@ -104,6 +104,16 @@ void HandshakeManager::DoHandshake(
   args_.args = channel_args;
   args_.event_engine = args_.args.GetObject<EventEngine>();
   args_.acceptor = acceptor;
+  // Add a channelz trace that we're performing a handshake.
+  // Note that we only commit this to the log if we see an error - otherwise
+  // it's ephemeral and is cleaned up when refs to it are released.
+  auto channelz_node = args_.args.GetObjectRef<channelz::BaseNode>();
```

Comment:
It looks like this setup moved from CallNextHandshakerLocked() to here.  I assume that's intended to combine log data from all handshakers into a single trace node, rather than have each handshaker be its own trace node?

Reply:
Yeah, it was a bit hard to follow